content/advent-2018/split-data-with-cdc.md

```diff
@@ -0,0 +1,232 @@
+
+Enough theory, let's dive into the code! 
+
+Splitting data
+==============
+
+The type which does all the hard work of computing the fingerprints over the sliding window is contained in the `Chunker` type. The Rabin fingerprint needs a so-called "irreducible polynomial" to work, which is encoded as an `uint64`. The `chunker` package exports the function `RandomPolynomial()` which creates a new polynomial for you to use. **Important:** Different polynomials will result in different fingerprints and therefore different boundaries for the chunks, so if you depend on your program computing the same boundaries, you'll need to pass in the same polynomial.
+
+Our first program will generate a random polynomial and create a new `Chunker` which reads data from `os.Stdin`:
+
+```go
+p, err := chunker.RandomPolynomial()
+if err != nil {
+	panic(err)
+}
+
+fmt.Printf("using polynomial %v for splitting data\n", p)
+
+chunker := chunker.New(os.Stdin, p)
+```
+
+By default, it tries to make chunks of one megabyte on average (you can control that with the method `SetAverageBits()`). Then we can repeatedly call the `Next()` function on the `Chunker` and pass in a byte slice buffer for the data read from the reader passed to `New()`, it'll return a `Chunk` type and an error:
+
+```go
+buf := make([]byte, 16*1024*1024) // 16 MiB
+for {
+	chunk, err := chunker.Next(buf)
+	if err == io.EOF {
+		break
+	}
+
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("%d\t%d\t%016x\n", chunk.Start, chunk.Length, chunk.Cut)
+}
+```
+
+Let's get some random data to test:
+
+```
+$ dd if=/dev/urandom bs=1M count=100 of=data
+100+0 records in
+100+0 records out
+104857600 bytes (105 MB, 100 MiB) copied, 0,548253 s, 191 MB/s
+```
+
+Now if we feed the data in this file to the program we've just written ([code](https://gist.github.com/fd0/a74e42a7a4f51c4ccaa6c11a09c14619)), it prints the start offset, length, and fingerprint for the chunk it found:
+
+```
+$ cat data | go run main.go
+using polynomial 0x3dea92648f6e83 for splitting data
+0       2908784 000c31839a100000
+2908784 2088949 001e8b4104900000
+4997733 1404824 000a18d1f0200000
+6402557 617070  001bc6ac84300000
+7019627 1278326 001734984b000000
+8297953 589913  0004cf802b600000
+8887866 554526  001eb5a362900000
+9442392 1307416 000ef1e549c00000
+[...]
+102769045       864607  00181b67df300000
+103633652       592134  0000a0c8b4200000
+104225786       631814  001d5ba20d38998b
+```
+
+On a second run, it'll generate a new polynomial and the chunks will be different. Let's fix the polynomial (`0x3dea92648f6e83`) for the next runs and change the program like this ([code](https://gist.github.com/bf984e1c40b56eaeff310d07a0d71128)):
```

Comment:
How about you move the **Important** note from the above to here? I think there's a strong connection between these two sentences since here you're "fixing" what has been described in the important part.